tests/api/v8/OAuth2Cest.php

```diff
@@ -51,15 +51,64 @@ public function TestScenarioInvalidClient(apiTester $I, \Helper\PhpBrowserDriver
         $I->seeResponseCodeIs(401);
     }
 
+    /**
+     * I want to make sure only the allowed grant types can be requested for any client
+     * @param apiTester $I
+     *
+     * HTTP Verb: POST
+     * URL: /api/oauth/access_token
+     */
+    public function TestScenarioGrantTypeNotAllowed(apiTester $I)
+    {
+        $client_id = 'API-ea74-c352-badd-c2be-5a8d9c9d4351';
+        $client_secret = 'secret';
+
+        $I->sendPOST(
+            $I->getInstanceURL().'/api/oauth/access_token',
+            array(
+                'grant_type' => 'password',
+                'client_id' => $client_id,
+                'client_secret' => $client_secret
+            )
+        );
+
+        $I->canSeeResponseIsJson();
+        $I->seeResponseCodeIs(400);
+    }
+
     /**
      * As a Rest API Client, I want to login so that I can get a JWT token
      * @param apiTester $I
      *
      * HTTP Verb: POST
      * URL: /api/oauth/access_token
      */
-    public function TestScenarioLogin(apiTester $I)
+    public function TestScenarioLoginWithPasswordGrant(apiTester $I)
+    {
+        $I->loginAsAdmin();
+    }
+
+    /**
+     * I want to be able to login with Client Credentials grant type
+     * @param apiTester $I
+     *
+     * HTTP Verb: POST
+     * URL: /api/oauth/access_token
+     */
+    public function TestScenarioLoginWithClientCredentialsGrant(apiTester $I)
     {
-       $I->loginAsAdmin();
+        $client_id = 'API-ea74-c352-badd-c2be-5a8d9c9d4351';
+        $client_secret = 'secret';
+
+        $I->sendPOST(
+            $I->getInstanceURL().'/api/oauth/access_token',
+            array(
+                'grant_type' => 'client_credentials',
+                'client_id' => $client_id,
+                'client_secret' => $client_secret
```

Comment:
To make it easier to reuse the client grant type in other tests, can we put this code in a method inside apiTester and then call it from here. Also consider using I-getClientId() and getClientSecret() here instead of hard coding the values.